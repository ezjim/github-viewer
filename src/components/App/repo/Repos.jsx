import React from 'react';
import PropTypes from 'prop-types';
import Repo from '../Repo/Repo.jsx';

const Repos = ({ repos }) => {
  const repoData = repos.map(repo => <Repo key={repo.id} {...repo} />);
  return (
    <ul>
      {repoData}
    </ul>
  );
};

Repos.propTypes = {
  repos: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    html_url: PropTypes.string.isRequired
  }))
};

export default Repos;
