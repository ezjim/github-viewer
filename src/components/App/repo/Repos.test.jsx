import React from 'react';
import { shallow } from 'enzyme';
import Repos from './Repos';

describe('Repos component', () => {
  it('renders Repos', () => {
    const repos = [
      {
        id: 233723653,
        name: 'about-me',
        html_url: 'https://github.com/ezjim/about-me'
      },
      {
        id: 236626883,
        name: 'bus-mall',
        html_url: 'https://github.com/ezjim/bus-mall'
      }
    ];
    const wrapper = shallow(<Repos repos={repos}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
