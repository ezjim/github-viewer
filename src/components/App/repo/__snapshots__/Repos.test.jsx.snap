// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos component renders Repos 1`] = `
<ul>
  <Repo
    html_url="https://github.com/ezjim/about-me"
    id={233723653}
    key="233723653"
    name="about-me"
  />
  <Repo
    html_url="https://github.com/ezjim/bus-mall"
    id={236626883}
    key="236626883"
    name="bus-mall"
  />
</ul>
`;
